docs/cli-commands/graphql.md

```diff
@@ -0,0 +1,501 @@
+| `--query`           | string | GraphQL query/mutation string or path to .graphql file         |
+| `--variables`       | string | Variables as JSON string or path to .json file                 |
+| `--operation`       | string | Execute named operation from DataHub's schema                  |
+| `--describe`        | string | Describe operation or type (searches both)                     |
+| `--recurse`         | flag   | Recursively explore nested types with --describe               |
+| `--list-operations` | flag   | List all available operations                                  |
+| `--list-queries`    | flag   | List available query operations                                |
+| `--list-mutations`  | flag   | List available mutation operations                             |
+| `--schema-path`     | string | Path to GraphQL schema files directory                         |
+| `--no-pretty`       | flag   | Disable pretty-printing of JSON output (default: pretty-print) |
+| `--format`          | choice | Output format: `human` (default) or `json` for LLM consumption |
+
+### Usage Patterns
+
+```shell
+# Discovery
+datahub graphql --list-operations
+datahub graphql --describe <name> [--recurse]
+
+# Execution
+datahub graphql --operation <name> [--variables <json>]
+datahub graphql --query <graphql> [--variables <json>]
+```
+
+## Advanced Examples
+
+### Complex Search with Filters
+
+```shell
+datahub graphql --operation searchAcrossEntities --variables '{
+  "input": {
+    "query": "customer",
+    "types": ["DATASET", "DASHBOARD"],
+    "filters": [{
+      "field": "platform",
+      "values": ["mysql", "postgres"]
+    }],
+    "start": 0,
+    "count": 20
+  }
+}'
+```
+
+### Adding Tags to Multiple Entities
+
+```shell
+# Add Important tag to a dataset
+datahub graphql --query 'mutation AddTag($input: TagAssociationInput!) {
+  addTag(input: $input)
+}' --variables '{
+  "input": {
+    "resourceUrn": "urn:li:dataset:(urn:li:dataPlatform:mysql,db.users,PROD)",
+    "tagUrn": "urn:li:tag:Important"
+  }
+}'
+```
+
+### Batch User Queries
+
+```shell
+# Get multiple users using raw GraphQL
+datahub graphql --query 'query GetUsers($urns: [String!]!) {
+  users: batchGet(urns: $urns) {
+    ... on CorpUser {
+      urn
+      username
+      properties {
+        email
+        displayName
+      }
+    }
+  }
+}' --variables '{"urns": ["urn:li:corpuser:alice", "urn:li:corpuser:bob"]}'
+```
+
+## Schema Introspection
+
+DataHub's GraphQL CLI uses intelligent introspection to understand the available operations and types. When introspection is unavailable, it falls back to parsing local GraphQL schema files.
+
+### Fallback Behavior
+
+1. **Introspection** (primary): Queries the live GraphQL endpoint
+2. **Schema Files** (fallback): Parses `.graphql` files from the schema directory
+3. **Hardcoded Operations** (last resort): Uses a minimal set of known operations
```

Comment:
Step 3 needs to be removed

And according to your last comment:
> - Respect user intent: --schema-path uses local files, no flag uses live introspection

There should be no fallback at all, right?